mm2src/coins/utxo/utxo_common.rs

```diff
@@ -382,27 +383,69 @@ pub async fn load_hd_accounts_from_storage(
     }
 }
 
+/// Requests balance of the given `address`.
 pub async fn address_balance<T>(coin: &T, address: &Address) -> BalanceResult<CoinBalance>
 where
     T: UtxoCommonOps + MarketCoinOps,
 {
+    if coin.as_ref().check_utxo_maturity {
+        let (mut unspents_map, _) = coin.get_mature_unspent_ordered_map(vec![address.clone()]).await?;
+        let unspents = unspents_map.remove(address).or_mm_err(|| {
```

Comment:
It seems that batch request usage is not required here. It also adds the code to handle error, which can't happen if the single request is used.

Reply:
Refactored